### PR TITLE
fix:  impossible to config RBAC if group name includes ','

### LINKED ISSUE #3012: It is impossible to config RBAC if group name includes ','
Checklist:

* [x] I've searched in the docs and FAQ for my answer: http://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

After [upgrading]( https://github.com/argoproj/argo-cd/pull/2517) to casbin to 1.9.1 it is impossible to have RBAC policy for a group that includes `,`. E.g. it is impossible to configure RBAC if OIDC returns something like `role,admin`: 

```
p, role,admin, projects, delete, *, allow
```

**Expected behavior**

It should be possible to put group name into quotes:

```
p, "role,admin", projects, delete, *, allow
```

### PR DESCRIPTION
**Pull request by alexmt**
Closes #3012

### CODE REVIEW

**Comment by jannfis** [comment]
Does this also fix #2992 by chance?

**Comment by alexmt** [comment]
I don't think so. The policy in #2992 is actually incorrect and argocd-server fails correctly. As I understand to close #2992 we need a tool that validates policy file.

**Comment by jannfis** [comment]
Hm, I would rather prefer to fail with an error (printed to log file of argocd-server) and then load default/built-in policy for cases in #2992, but let's take this discussion to the appropriate issue.

File: util/rbac/rbac_test.go [hunk 1]
```diff
@@ -371,6 +371,8 @@ func TestValidatePolicy(t *testing.T) {
 	goodPolicies := []string{
 		"p, role:admin, projects, delete, *, allow",
 		"",
+		"#",
+		`p, "role,admin", projects, delete, *, allow`,
```
Comment (jannfis) [inline util/rbac/rbac_test.go]: We should also test for strings starting with whitespace.

File: util/rbac/rbac.go [hunk 2]
```diff
@@ -273,12 +273,33 @@ func (a *argocdAdapter) LoadPolicy(model model.Model) error {
...
+	reader := csv.NewReader(strings.NewReader(line))
+	reader.TrimLeadingSpace = true
+	tokens, err := reader.Read()
+	if err != nil {
+		return err
+	}
+
+	key := tokens[0]
+	sec := key[:1]
+	model[sec][key].Policy = append(model[sec][key].Policy, tokens[1:])
```
Comment (jannfis) [inline util/rbac/rbac.go]: Can we please add some clarification what `key` and `sec` actually represent in this context? Like, a little comment would be nice :)

**Review by jannfis — CHANGES_REQUESTED** [review]

File: util/rbac/rbac_test.go (hunk 1 above)
Comment (alexmt) [reply util/rbac/rbac_test.go]: done

File: util/rbac/rbac.go [hunk 3]
```diff
@@ -273,12 +273,33 @@ func (a *argocdAdapter) LoadPolicy(model model.Model) error {
 			if line == "" {
 				continue
 			}
-			persist.LoadPolicyLine(line, model)
+			if err := loadPolicyLine(line, model); err != nil {
```
[the remark this answers is not in the document]
Comment (alexmt) [reply util/rbac/rbac.go]: done

File: util/rbac/rbac.go (hunk 2 above)
Comment (alexmt) [reply util/rbac/rbac.go]: to be honest I don't know what is sec and key here :).  The function is a copy-paste from https://github.com/casbin/casbin/blob/master/persist/adapter.go#L24 with improved parsing that uses csv reader. Added comment to the function definition.

**Review by jannfis — APPROVED** [review]
